File: Back_end/src/main/java/com/example/project_piatt/Model/RoomDTO.java
package com.example.project_piatt.Model;

import com.example.project_piatt.Entity.Booking;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class RoomDTO {
    private Long id;
    private String nome;
    private String tipo;
    private String descrizione;
    private int prezzo;
    private String imageName;
    //private int modifica;
  //  private int version;
    private String persone;
   // private List<BookingDTO> bookingDTOList;
}

File: Front_end/lib/model/RoomDTO.dart
import 'package:flutter_application_1/model/BookingDTO.dart';

class RoomDTO {
  final int? id;
  final String? nome;
  final String? tipo;
  final String? descrizione;
  final double? prezzo;
  final String? imageName;
  final List<BookingDTO> bookingDTOList;

  RoomDTO({
    this.id,
    this.nome,
    this.tipo,
    this.descrizione,
    this.prezzo,
    this.imageName,
    List<BookingDTO>? bookingDTOList, // Optional and nullable
  }) : bookingDTOList = bookingDTOList ?? [];  // Se è null, assegna una lista vuota

  factory RoomDTO.fromJson(Map<String, dynamic> json) {
    return RoomDTO(
      id: json['id'] as int?,
      nome: json['nome'] as String?,
      tipo: json['tipo'] as String?,
      descrizione: json['descrizione'] as String?,
      prezzo: json['prezzo'] as double?,
      imageName: json['imageName'] as String?,
      bookingDTOList: json['bookingDTOList'] != null
          ? List<BookingDTO>.from(json['bookingDTOList'].map((x) => BookingDTO.fromJson(x)))
          : [], // Se bookingDTOList è null, assegna una lista vuota
    );
  }
}
